lib/facter/resolvers/hostname_resolver.rb

```diff
@@ -42,7 +42,12 @@ def retrieve_from_fqdn(output)
         end
 
         def retrieve_with_addrinfo(host)
-          name = Socket.getaddrinfo(host, 0, Socket::AF_UNSPEC, Socket::SOCK_STREAM, nil, Socket::AI_CANONNAME)[0]
+          begin
+            name = Socket.getaddrinfo(host, 0, Socket::AF_UNSPEC, Socket::SOCK_STREAM, nil, Socket::AI_CANONNAME)[0]
+          rescue SocketError => e
```

Comment:
Thanks, this solves the issue I was seeing, but one question, can `getaddrinfo` raise exceptions other than `SocketError`? Should we do `rescue => e` instead?